src/main/java/bwcrossmod/galacticgreg/MTEVoidMiners.java

```diff
@@ -13,66 +13,325 @@
+                .addInfo("Put the Ore into the input bus to set the Whitelist/Blacklist")
+                .addInfo("Use a screwdriver to toggle Whitelist/Blacklist")
+                .addInfo(
+                    "Blacklist or non Whitelist Ore will be " + EnumChatFormatting.DARK_RED
+                        + "VOIDED"
+                        + EnumChatFormatting.RESET
+                        + ".")
+                .beginStructureBlock(7, 9, 7, false)
+                .addStructureInfo(
+                    EnumChatFormatting.GOLD + "20x" + EnumChatFormatting.GRAY + " Mining Osmiridium Casing")
+                .addStructureInfo(EnumChatFormatting.GOLD + "42x" + EnumChatFormatting.GRAY + " Osmiridium Frame Box")
+                .addStructureInfo(
+                    EnumChatFormatting.GOLD + "6x" + EnumChatFormatting.GRAY + " Bolted Osmiridium Casing")
+                .addStructureInfo(
+                    EnumChatFormatting.GOLD + "6x" + EnumChatFormatting.GRAY + " Rebolted Osmiridium Casing")
+                .addEnergyHatch(VN[this.getMinTier()] + "+, Any base casing")
+                .addMaintenanceHatch("Any base casing")
+                .addInputBus("Mining Pipes or Ores, optional, any base casing")
+                .addInputHatch("Optional noble gas, any base casing")
+                .addOutputBus("Any base casing")
+                .toolTipFinisher();
+            return tt;
+        }
+
+        public void construct(ItemStack stackSize, boolean hintsOnly) {
+            buildPiece(STRUCTURE_PIECE_MAIN, stackSize, hintsOnly, 3, 7, 1);
+        }
+
+        public int survivalConstruct(ItemStack stackSize, int elementBudget, ISurvivalBuildEnvironment env) {
+            if (mMachine) return -1;
+            return survivialBuildPiece(STRUCTURE_PIECE_MAIN, stackSize, 3, 7, 1, elementBudget, env, false, true);
         }
 
         @Override
-        protected Materials getFrameMaterial() {
-            return Materials.Europium;
+        public boolean checkMachine(IGregTechTileEntity aBaseMetaTileEntity, ItemStack aStack) {
+            return (checkPiece(STRUCTURE_PIECE_MAIN, 3, 7, 1)
+                // old structure (to be deprecated on future version)
+                || (checkPiece(STRUCTURE_PIECE_OLD, 1, 6, 0))) && checkHatches()
+                && GTUtility.getTier(getMaxInputVoltage()) >= getMinTier()
+                && mMaintenanceHatches.size() == 1;
+        }
+
+        public VMLUV(String aName, int tier) {
+            super(aName, tier);
+            casingTextureIndex = ((BlockCasings4) GregTechAPI.sBlockCasings4).getTextureIndex(14);
+        }
+
+        public VMLUV(int aID, String aName, String aNameRegional) {
+            super(aID, aName, aNameRegional, 1);
         }
 
         @Override
-        protected int getCasingTextureIndex() {
-            return 8;
+        protected int getControllerTextureIndex() {
+            return ((BlockCasings4) GregTechAPI.sBlockCasings4).getTextureIndex(14);
```

Comment:
nitpick: Can you switch this to the Casing?
```suggestion
            return Casings.MiningOsmiridiumCasing.getTextureId();
```